src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -140,7 +140,9 @@ private async Task Initialize(CancellationToken cancellationToken)
         {
             SyncReport syncReport = new(_api.SyncPeerPool!, _api.NodeStatsManager!, _api.SyncModeSelector, _syncConfig, _api.Pivot, _api.LogManager);
 
-            _api.BlockDownloaderFactory ??= new BlockDownloaderFactory(_api.SpecProvider!,
+            _api.BlockDownloaderFactory ??= new BlockDownloaderFactory(
+                _syncConfig.MaxProcessingThread,
```

Comment:
Would put it at the end, so you can specify 0 as default - useful for testing and others when you can just omit the parameter. Makes the default more explicit and could significantly reduce the number of changed files in this PR.

Reply:
I put it in front so that I don't miss anything. I'll refactor this later to reduce the need of such large unnecessary modification again.

Reply:
Later - in this PR?

Reply:
No, other PR.

Reply:
:(